{{cookiecutter.project_slug}}/clients/react/src/pages/LogIn.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useContext } from "react"
+import {
+  Heading,
+  Input,
+  Text,
+  Grid,
+  GridItem,
+  Button,
+  VStack,
+  Link,
+  Image,
+  Show,
+  Hide,
+  FormControl,
+  FormErrorMessage,
+} from "@chakra-ui/react"
+import { useMutation } from "@apollo/client"
+import { LOG_IN } from "../utils/mutations"
+import { AuthContext } from "../utils/auth"
+import { useNavigate } from "react-router-dom"
+import Logo from "src/assets/images/logo.svg"
+
+export default function LogIn() {
+  const [email, setEmail] = useState("")
+  const [password, setPassword] = useState("")
+  const [error, setError] = useState(false)
+
+  const { updateToken } = useContext(AuthContext)
+
+  let navigate = useNavigate()
+  const [logIn] = useMutation(LOG_IN, {
+    onCompleted: (data: any) => {
+      localStorage.setItem("auth-token", data.tokenAuth.token)
+      updateToken(data.tokenAuth.token)
+
+      navigate("/verify")
+    },
+    onError: (error: any) => {
+      if (error.message === "Please enter valid credentials") {
+        setError(true)
+      }
+    },
+  })
+
+  const handleLogin = () => {
+    logIn({
+      variables: {
+        email,
+        password,
+      },
+    })
+  }
+
+  return (
+    <Grid
+      templateColumns={{
+        base: "repeat(1, 1fr)",
+        md: "1fr 350px 350px 1fr",
+      }}
+      templateRows={{ base: "repeat(2, 2fr)" }}
+    >
+      <GridItem colStart={2} colEnd={3}>
+        <VStack
+          justify="center"
+          align="center"
+          h={{ base: "80vh", md: "50vh" }}
+          mx={5}
+          mt={{ base: 0, md: "30%" }}
+        >
+          <Show breakpoint="(max-width: 48em)">
+            <Image maxW={20} src={Logo} />
+          </Show>
+          <Heading mt={5}>PORTAL LOG IN</Heading>
+          <Text>Enter your login credentials below</Text>
+          <FormControl isInvalid={error}>
+            <Input
+              placeholder="Email"
+              onChange={(e) => {
+                setEmail(e.target.value)
+              }}
+              my={5}
```

Comment:
For these to be fully-controlled components, do we not want a prop of
```jsx
<Input value={email} />
```

and same for the password input?

Reply:
good catch, fixed!